src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs

```diff
@@ -26,61 +26,100 @@ public partial class ConsoleLogs : ComponentBase, IAsyncDisposable
     private static ViewModelMonitor<ResourceViewModel> GetViewModelMonitor(IDashboardViewModelService dashboardViewModelService)
         => dashboardViewModelService.GetResources();
 
-    private FluentSelect<ResourceViewModel>? _resourceSelectComponent;
+    private FluentSelect<Option<string>>? _resourceSelectComponent;
+    private Option<string>? _selectedOption;
     private ResourceViewModel? _selectedResource;
     private readonly Dictionary<string, ResourceViewModel> _resourceNameMapping = new();
-    private IEnumerable<ResourceViewModel> Resources => _resourceNameMapping.Select(kvp => kvp.Value).OrderBy(c => c.Name);
+    private List<Option<string>>? Resources { get; set; }
     private LogViewer? _logViewer;
     private readonly CancellationTokenSource _watchContainersTokenSource = new();
     private CancellationTokenSource? _watchLogsTokenSource;
     private string _status = LogStatus.Initializing;
 
+    private readonly TaskCompletionSource _renderCompleted = new();
```

Comment:
Just append TaskCompletionSource or something. The name looks like a bool otherwise.